src/model/ItemMovement.js

```diff
@@ -0,0 +1,675 @@
+			}
+		}
+		else {
+			ty = utils.pointOnPlat(platform, tx).y;
+		}
+		this.kickStage = 3;
+		this.item.apiStartFlyingTo(tx, ty, 90, 'movementCallback');
+	}
+	else {
+		delete this.kickStage;
+		delete this.kickVx;
+		this.item.onPlatformLanding();
+	}
+};
+
+/**
+ * Builds the path for an item.
+ *
+ * @param {string} transport the transport type for the movement
+ * @param {object} dest destination for the movement
+ * @param {object} options options passed into movement
+ * @returns {object} the pathing object for the movement
+ */
+ItemMovement.prototype.buildPath = function buildPath(transport, dest, options) {
+	var path;
+	if (transport === 'walking') {
+		//TODO: Actual pathing
+		path = [
+			{x: dest.x, y: dest.y, transport: 'walking'}
+		];
+	}
+	else if (transport === 'direct') {
+	// Build a streight line movement path
+		path = [
+			{x: dest.x,
+			 y: dest.y,
+			 speed: options.speed,
+			 transport: 'direct'}
+		];
+	}
+	else if (transport === 'flying') {
+		if (!options.stopAtEnd) {
+			path = [
+			 {left: dest.left,
+			  right: dest.right,
+			  width: dest.width,
+			  top: dest.top,
+			  height: dest.height,
+			  speed: options.speed,
+			  stopAtEnd: false,
+			  transport: 'flying'}
+			];
+		}
+		else {
+			path = [
+			 {x: dest.x,
+			  y: dest.y,
+			  speed: options.speed,
+			  stopAtEnd: true,
+			  transport: 'flying'}
+			];
+		}
+	}
+	else if (transport === 'kicked') {
+		this.kickVx = options.vx;
+		this.kickStage = 1;
+		this.movementCallback = this.kickCallback;
+		path = [
+			 {x: this.item.x + options.vx,
+			  y: this.item.y + options.vy,
+			  speed: 90,
+			  stopAtEnd: true,
+			  transport: 'flying'}
+		];
+	}
+	return path;
+};
+
+
+
+/**
+ * These are functions that will be called by the base Item to envoke
+ * movement.
+ */
+
+/**
+ * An item has gotten a movement request.
+ *
+ * @param {string} transport the transportation for this movement
+ * @param {object} dest destination for the movement
+ * @param {object} options for this movement
+ * @returns {boolean} true if movement is possible and started
+*/
+ItemMovement.prototype.startMove = function startMove(transport, dest, options) {
+	if (this.path) {
+		this.stopMove(MOVE_CB_STATUS.STOP_NEW_MOVE);
+	}
+	this.flags = options;
+	if ('callback' in options) {
+		this.callback = this.item[options.callback];
+	}
+	this.facing = 0; // unset
+	if (options.path) {
+		this.path = options.path;
+	}
+	else {
+		this.path = this.buildPath(transport, dest, options);
+	}
+	this.moveStep();
+	// Check if we are stuck, i.e. completely unable to move.
+	if (!this.path || this.path.length === 0) {
+		return false;
+	}
+	this.item.setGsTimer({fname: 'movementTimer', delay: 333, interval: true,
+		internal: true});
+	return true;
+};
+
+/*
+ * Stop item movement.
+ */
+ItemMovement.prototype.stopMovement = function stopMovement() {
+	log.debug('%s.apiStopMoving()', this);
+	this.stopMove(MOVE_CB_STATUS.STOP);
+	return true;
```

Comment:
That return is unnecessary AFAICS (at least `Item.prototype.stopMoving` does not use it)